src/schema.rs

```diff
@@ -294,16 +295,33 @@ pub enum RecordFieldOrder {
 
 impl RecordField {
     /// Parse a `serde_json::Value` into a `RecordField`.
-    fn parse(field: &Map<String, Value>, position: usize) -> Result<Self, Error> {
+    fn parse(field: &Map<String, Value>, position: usize, parsed_record: &mut HashMap<String, Schema>) -> Result<Self, Error> {
```

Comment:
I am not sure I like this additional argument that we have to carry around everywhere. It is pretty obscure and it's adding quite a bit of complexity (and `clone`)...